### PR TITLE
Toasts (Notification Component)

### PR DESCRIPTION
**Pull request by iMADi-ARCH**
- Improved ui for auth pages.
- group similar routes together with `(name)` file convention as per next js [route groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups).
- Toasts which notify user of success, warning or error messages when doing auth / upload.

### CODE REVIEW

File: frontend/app/components/ui/Toast.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    const [count, setCount] = useState(0);
+
+    useImperativeHandle(forwardedRef, () => ({
+      publish: () => setCount((count) => count + 1),
+    }));
+
+    return (
+      <>
+        <ToastPrimitive.Provider duration={Infinity}>
+          {Array.from({ length: count }).map((_, index) => (
```
Comment (mamadoudicko) [inline frontend/app/components/ui/Toast.tsx]: Some remarks:

1. As new toasts are added, all indices (and thus keys) change, causing unnecessary re-rendering of all toasts, leading to performance inefficiencies.

2. If in the future you decide to allow toasts to be removed individually, using indices as keys could lead to incorrect behavior or bugs due to shifted indices.

3. React uses keys for diffing and deciding which components to update, thus if the count changes, using indices could lead to incorrect diff results, causing potential issues with state updates.

Proposition:

```ts
useImperativeHandle(forwardedRef, () => ({
  publish: () => setToasts((toasts) => [...toasts, { id: generateUniqueID(), content: children }]),
}));
```
```ts
 key={toast.id}
```

**Review by mamadoudicko — COMMENTED** [review]
Amazing job on this pull request! I've spotted a couple of minor points for consideration, but let's be clear - they don't overshadow the brilliance of your work. Outstanding job!

**Comment by StanGirard** [comment]
@mamadoudicko last but not least  ;) 
image

File: frontend/app/components/NavBar/NavItems.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { FC, HTMLAttributes } from "react";
+import DarkModeToggle from "./DarkModeToggle";
+import Button from "../ui/Button";
+
+interface NavItemsProps extends HTMLAttributes<HTMLUListElement> {}
+
+const NavItems: FC<NavItemsProps> = ({ className, ...props }) => {
+  return (
+    // <div className={cn("flex flex-1 items-center", className)} {...props}>
+    <ul
```
[the remark this answers is not in the document]
Comment (iMADi-ARCH) [reply frontend/app/components/NavBar/NavItems.tsx]: done!

**Review by StanGirard — APPROVED** [review]

File: frontend/app/components/ui/Toast.tsx (hunk 1 above)
Comment (iMADi-ARCH) [reply frontend/app/components/ui/Toast.tsx]: Toasts now get different ids when published which are generated by this function in `@/lib/utils.ts` file.

```ts
export function generateUniqueId() {
  const timestamp = Date.now();
  const random = Math.floor(Math.random() * 10000);
  return `${timestamp}-${random}`;
}
```

**Comment by StanGirard** [comment]
Amazing @iMADi-ARCH ! Thanks !

**Pull request merged**